src/query/api/v1/handler/placement/replace.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package placement
+
+import (
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/m3db/m3/src/cluster/placement"
+	"github.com/m3db/m3/src/query/api/v1/handler"
+	"github.com/m3db/m3/src/query/generated/proto/admin"
+	"github.com/m3db/m3/src/query/util/logging"
+	xhttp "github.com/m3db/m3/src/x/net/http"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"go.uber.org/zap"
+)
+
+const (
+	// ReplaceHTTPMethod is the HTTP method for the the replace endpoint.
+	ReplaceHTTPMethod = http.MethodPost
+
+	replacePathName = "replace"
+)
+
+var (
+	// M3DBReplaceURL is the url for the m3db replace handler (method POST).
+	M3DBReplaceURL = path.Join(handler.RoutePrefixV1, M3DBServicePlacementPathName, replacePathName)
```

Comment:
You wrote documentation for m3db/m3coordiantor/m3agg but only created a route for M3DB. Wanna create the other routes? Also we should get @cw9 to confirm there are no other special changes we need to make to support M3Agg replaces (when I did the m3agg stuff I looked at the r2admin code to see if there was anything special but Chao can probably just answer this question quickly)

Reply:
Will do

Reply:
You might need to change `ServiceWithAlgo` function to make sure it's using the same logic with r2admin, in there we had functions like `GetOptionsForReplace`, Richies work probably only moved  `GetOptions` over.

Reply:
@cw9 do we need more logic past what's here? https://github.com/m3db/m3/blob/a2fab37bae1b9144354eba854bcb050a7d6398b3/src/query/api/v1/handler/placement/common.go#L262